### PR TITLE
Fix bug when rewriting measurement type ops.

### PR DESCRIPTION
**Pull request by schweitzpgi**
Measurements now thread the wires as well as the bool result value in the value semantics. Each wire needs to be forwarded correctly.

### CODE REVIEW

File: lib/Optimizer/Transforms/RegToMem.cpp [hunk 1]
```diff
@@ -143,14 +143,19 @@ struct RegToMemAnalysis {
       }
     });
     func.walk([&](Operation *op) {
+      // FIXME: This really needs to figure out which operands are wires and
+      // which are not. It blindly assumes everything is a wire.
```
Comment (bmhowe23) [inline lib/Optimizer/Transforms/RegToMem.cpp:147]: Should this be captured in a new issue?

**Review by bmhowe23 — APPROVED** [review]
LGTM. We've discussed offline that we expect the mapping changes to exercise and test this new code. I'm just reiterating that here in the review so that we have a record of it (as a partial explanation for why there's not currently any new tests for this PR).

File: lib/Optimizer/Transforms/RegToMem.cpp (hunk 1 above)
Comment (schweitzpgi) [reply lib/Optimizer/Transforms/RegToMem.cpp:147]: Yes.

Comment (schweitzpgi) [reply lib/Optimizer/Transforms/RegToMem.cpp:147]: https://github.com/NVIDIA/cuda-quantum/issues/763

**Pull request merged**